src/pfs/server/api_server.go

```diff
@@ -77,11 +77,28 @@ func (a *apiServer) InspectRepo(ctx context.Context, request *pfs.InspectRepoReq
 	a.versionLock.RLock()
 	defer a.versionLock.RUnlock()
 	ctx = versionToContext(a.version, ctx)
-	clientConn, err := a.getClientConn(a.version)
+
+	clientConns, err := a.router.GetAllClientConns(a.version)
 	if err != nil {
 		return nil, err
 	}
-	return pfs.NewInternalAPIClient(clientConn).InspectRepo(ctx, request)
+
+	var repoInfos []*pfs.RepoInfo
+	for _, clientConn := range clientConns {
+		repoInfo, err := pfs.NewInternalAPIClient(clientConn).InspectRepo(ctx, request)
+		if err != nil {
+			return nil, err
+		}
+		repoInfos = append(repoInfos, repoInfo)
+	}
+
+	for _, repoInfo := range pfs.ReduceRepoInfos(repoInfos) {
+		if repoInfo.Repo.Name == request.Repo.Name {
+			 return repoInfo, nil
+		}
+	}
+
+	return nil, fmt.Errorf("internal API servers returned the wrong repo info")
```

Comment:
For errors like this I like to say something like:

`"incorrect repo returned (this is likely a bug)"`

It makes it clear that this is an internal issue and not an input issue and will hopefully prompt the user to open a GH issue. I worry that mentioning things like internal API servers will be scary to them.